prov/verbs/src/verbs_info.c

```diff
@@ -1375,6 +1415,31 @@ static int fi_ibv_handle_sock_addr(const char *node, const char *service,
 	return ret;
 }
 
+static int fi_ibv_port_is_set(void *addr)
+{
+	if (!addr)
+		return 0;
+
+	switch (ofi_sa_family(addr)) {
+	case AF_INET:
+		return ofi_sin_port(addr) ? 1 : 0;
+	case AF_INET6:
+		return ofi_sin6_port(addr) ? 1 : 0;
+	default:
+		VERBS_WARN(FI_LOG_FABRIC, "Unknown address format\n");
+		return 0;
+	}
+}
+
+/* TODO revisit this if and when getinfo is refactored */
+static int fi_ibv_is_any_addr(const char *node, const char *service, const struct fi_info *hints)
+{
+	return (!node && (!hints || !hints->dest_addr) &&
+		((hints && ofi_is_any_addr(hints->src_addr) &&
+		  fi_ibv_port_is_set(hints->src_addr)) || service));
```

Comment:
It still seems wrong to associate having a port with an address being a wildcard.  A port or service doesn't associate a request with any specific device or fabric.